ext/semian/tickets.c

```diff
@@ -1,12 +1,51 @@
 #include "tickets.h"
 
-static void
-initialize_tickets(int sem_id, int tickets);
+// Update the ticket count for static ticket tracking
+static VALUE
+update_ticket_count(update_ticket_count_t *tc);
 
-static void
-configure_static_tickets(int sem_id, int tickets);
+static int
+calculate_quota_tickets(int sem_id, double quota);
 
-VALUE
+void
+configure_tickets(int sem_id, int tickets, double quota)
+{
+  int state = 0;
+  update_ticket_count_t tc;
+
+  if (quota > 0) {
+    tickets = calculate_quota_tickets(sem_id, quota);
+  }
+
+  /*
+    A manually specified ticket count of 0 is special, meaning "don't set"
+    We need to throw an error if we set it to 0 during initialization.
+    Otherwise, we back out of here completely.
+  */
+  if (get_sem_val(sem_id, SI_SEM_CONFIGURED_TICKETS) == 0 && tickets == 0) {
```

Comment:
this is a bit ugly, but I think it's the best spot to put this check in @sirupsen